@here/harp-mapview/lib/ScreenCollisions.ts

```diff
@@ -140,6 +172,44 @@ export class ScreenCollisions {
         return this.screenBounds.intersects(bounds);
     }
 
+    /**
+     * Checks if the given screen bounds is contained within the frustum of the active camera.
+     *
+     * @param bounds The bounding box in world coordinates.
+     */
+    isFullyVisible(bounds: Math2D.Box): boolean {
+        return this.screenBounds.containsBox(bounds);
+    }
+
+    /**
+     * Test whether a given [[CollisionBox]] intersects with any of the details in the specified
+     * [[IBox]]es.
+     *
+     * @param testBox The box to test for intersection.
+     * @param boxes The candidate boxes the test box may intersect with. It's assumed that the
+     * global bounds of these boxes intersect with the given test box.
+     * @returns `true` if any intersection found.
+     */
+    intersectsDetails(testBox: CollisionBox, boxes: IBox[]): boolean {
+        for (const box of boxes) {
+            if (box instanceof DetailedCollisionBox) {
+                for (const detailBox of box.detailBoxes) {
+                    if (detailBox.intersects(testBox)) {
```

Comment:
Some definitions before asking:
* C ... course bounding box of the line label that we want to add (i.e. `testBox`)
* c_i ... detailed glyph bounding boxes of the line that we want to add (i.e. `testBox.detailBoxes`)
* A_i ... course bounding boxes of all the already allocated labels that intersect with C (i.e. `boxes`)
* A_i_j ... detailed glyph bounding boxes of the already allocated labels (i.e. `detailBox`)

I would assume that the algorithm checks all A_i_j versus all c_i but as far as I can see you only check all A_i_j versus C or is `detailBox.intersects(testBox)` doing some magic?